support/util/deployment.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func IsDeploymentReady(ctx context.Context, c crclient.Client, deployment *appsv1.Deployment) (bool, error) {
```

Comment:
Orthogonal to this PR I think we should review this pattern where we let the cpo use a top level library. This might  mean a CPO backport fix will enforce an unnecessary backport to the hypershift operator as well. cc @alvaroaleman @csrwng @sjenning @ironcladlou